test/data/testNMR.py

```diff
@@ -17,14 +19,60 @@
 
 class GenericNMRTest(unittest.TestCase):
     """Generic NMR unittest"""
+    
+    def testkeys(self):
+        """Check dictionary keys are ints."""
+        key_types = set([type(key) for key in self.data.nmrtensors.keys()])
+        assert len(key_types) == 1 and int in key_types
 
     def testsize(self):
         """Check to make sure there are the correct number of tensors parsed"""
         assert len(self.data.nmrtensors) == self.data.natom
-        assert len(self.data.nmrtensors[0]) == 3
+        assert len(self.data.nmrtensors[0]) == 4
         assert self.data.nmrtensors[0]["total"].shape == (3, 3)
+        
+    def testisotropic(self):
+        """Check the total isotropic value matches the computed value."""
+        tensor = self.data.nmrtensors[0]
+        total = 0.0
+        for t_type in ("diamagnetic", "paramagnetic"):
+            total += sum(numpy.linalg.eigvals(tensor[t_type])) / len(numpy.linalg.eigvals(tensor[t_type]))
+        
+        assert total == pytest.approx(tensor['isotropic'], abs = 3)
 
 
+class GenericNMRCouplingTest(unittest.TestCase):
+    """Generic NMR spin-spin coupling unittest"""
+    
+    def testkeys(self):
+        """Check dictionary keys are ints."""
+        assert all(set([all((type(key[0]) == int, type(key[1]) == int)) for key in self.data.nmrcouplingtensors.keys()]))
+        
+        assert all(
+            [
+                all((type(isotropekey[0]) == int, type(isotropekey[1]) == int))
```

Comment:
isotrope -> isotope

Reply:
I knew I was going to make this mistake eventually